component/http/middleware.go

```diff
@@ -153,6 +159,76 @@ func ignore(ignoreRoutes []string, url string) bool {
 	return false
 }
 
+func parseAcceptEncoding(header string) (string, error) {
+	if header == anythingHeader {
+		return anythingHeader, nil
+	}
+
+	weighted := make(map[float64]string)
+
+	addWithWeight := func(weight float64, algorithm string) {
```

Comment:
Do we need this function? it is just a one liner and having a local function and another function with the real name is confusing and not very clean.

Reply:
Agreed. It sneaked in after extraction of the function it's calling.